src/tike/lamino/lamino_bucket.py

```diff
@@ -0,0 +1,185 @@
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS       #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL UChicago     #
+# Argonne, LLC OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,        #
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,    #
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;        #
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER        #
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT      #
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN       #
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE         #
+# POSSIBILITY OF SUCH DAMAGE.                                             #
+# #########################################################################
+
+__author__ = "Daniel Ching, Viktor Nikitin, Xiaodong Yu"
+__copyright__ = "Copyright (c) 2020, UChicago Argonne, LLC."
+__docformat__ = 'restructuredtext en'
+__all__ = [
+    "reconstruct",
+    "simulate",
+]
+
+import logging
+import numpy as np
+
+from tike.communicators import Comm
+from tike.operators import Bucket as Lamino
+from tike.lamino import solvers
+
+logger = logging.getLogger(__name__)
+
+
+def simulate(
+        obj,
+        theta,
+        tilt,
+        **kwargs
+):  # yapf: disable
+    """Return complex values of simulated laminography data."""
+    assert obj.ndim == 3
+    assert theta.ndim == 1
+    with Lamino(
+            n=obj.shape[-1],
+            tilt=tilt,
+            **kwargs,
+    ) as operator:
+        grid = operator._make_grid().reshape(obj.shape[-1]**3, 3)
+        data = operator.fwd(
+            u=operator.asarray(obj, dtype='complex64'),
+            theta=operator.asarray(theta, dtype='float32'),
+            grid=operator.asarray(grid, dtype='int16'),
+        )
+        assert data.dtype == 'complex64', data.dtype
+        return operator.asnumpy(data)
+
+
+def reconstruct(
+        data,
+        theta,
+        tilt,
+        algorithm,
+        obj=None, num_iter=1, rtol=-1, eps=1e-3,
+        num_gpu=1,
+        obj_split=1,
+        **kwargs
+):  # yapf: disable
+    """Solve the Laminography problem using the given `algorithm`.
+
+    Parameters
+    ----------
+    algorithm : string
+        The name of one algorithms from :py:mod:`.lamino.solvers`.
+    rtol : float
+        Terminate early if the relative decrease of the cost function is
+        less than this amount.
+
+    """
+    n = data.shape[2]
+    obj = np.zeros([n, n, n], dtype='complex64') if obj is None else obj
+    if algorithm in solvers.__all__:
+        # Initialize an operator.
+        with Lamino(
+                n=obj.shape[-1],
+                tilt=tilt,
+                eps=eps,
+                **kwargs,
+        ) as operator, Comm(num_gpu, mpi=None) as comm:
+            # send any array-likes to device
+            data_split = comm.pool.num_workers // obj_split
```

Comment:
Tests are failing because `data_split = zero` when `num_workers = 1` and `obj_split > 1`. We need to enforce `obj_split <= num_workers` or something that prevents splitting data into zero groups.